### PR TITLE
Feature/auth security

### PR DESCRIPTION
**Pull request by OvidiuGi**

### CODE REVIEW

File: src/Repository/ProgrammeRepository.php [hunk 1]
```diff
@@ -99,8 +110,9 @@ public function getPaginatedFilteredSorted(
             $direction = 'ASC';
         }
 
-        if ('' != $sort) {
-            $query = $query->orderBy("p.$sort", $direction);
+        if (null != $sort) {
+            $query
```
Comment (live-happy-code-hard) [inline src/Repository/ProgrammeRepository.php:114]: Can be written on a single line

File: src/Repository/ProgrammeRepository.php [hunk 2]
```diff
@@ -77,20 +88,20 @@ public function getPaginated(int $page, int $limit): array
...
-            ->setMaxResults($paginate['maxPerPage']);
+            ->setFirstResult($paginate['size'] * ($paginate['page'] - 1))
+            ->setMaxResults($paginate['size']);
 
         foreach ($filters as $key => $value) {
-            if ('' != $value) {
-                $query = $query->where("p.$key = :$key");
-                $query->setParameter(':key', $value);
+            if (null != $value) {
+                $query->where("p.$key = :value");
```
Comment (live-happy-code-hard) [inline src/Repository/ProgrammeRepository.php:103]: Is there a validation of the $key (the filter field) done anywhere? Unless that is done, this code is dangerous.

File: src/Repository/ProgrammeRepository.php [hunk 3]
```diff
@@ -99,8 +110,9 @@ public function getPaginatedFilteredSorted(
             $direction = 'ASC';
         }
 
-        if ('' != $sort) {
-            $query = $query->orderBy("p.$sort", $direction);
```
Comment (live-happy-code-hard) [inline src/Repository/ProgrammeRepository.php:103]: is there a validation of the value in $sort done anywhere? It should be validated that is a valid field that can be sorted upon. Unless there is such a validation, this code is dangerous.

File: src/Repository/ProgrammeRepository.php (hunk 2 above)
Comment (OvidiuGi) [reply src/Repository/ProgrammeRepository.php:103]: It wasn't, but i added a validation.

[the remark this answers is not in the document]
Comment (OvidiuGi) [reply src/Repository/ProgrammeRepository.php]: Yes, it printed the wrong results but i fixed it.

File: src/Repository/ProgrammeRepository.php (hunk 1 above)
Comment (OvidiuGi) [reply src/Repository/ProgrammeRepository.php:115]: The if statement or the code inside of it?

File: src/Repository/ProgrammeRepository.php [hunk 4]
```diff
@@ -77,20 +89,20 @@ public function getPaginated(int $page, int $limit): array
...
-            ->setMaxResults($paginate['maxPerPage']);
+            ->setFirstResult($paginate['size'] * ($paginate['page'] - 1))
+            ->setMaxResults($paginate['size']);
 
         foreach ($filters as $key => $value) {
-            if ('' != $value) {
-                $query = $query->where("p.$key = :$key");
-                $query->setParameter(':key', $value);
+            if (in_array($key, self::PROGRAMME_FIELDS) && null != $value) {
+                $query->andWhere("p.$key LIKE :value");
```
Comment (live-happy-code-hard) [inline src/Repository/ProgrammeRepository.php:104]: Are you sure you can use the LIKE operator with all those PROGRAMME_FIELDS? Not all of them hold string values, and some don't make sense to be used with LIKE. Also :value remains the same between filters, have you checked there is no overwriting of the parameter value?

File: src/Repository/ProgrammeRepository.php [hunk 5]
```diff
@@ -77,20 +89,20 @@ public function getPaginated(int $page, int $limit): array
...
+                $query->setParameter(':value', '%' . $value . '%');
```
Comment (live-happy-code-hard) [inline src/Repository/ProgrammeRepository.php:105]: The % and % should be put in the line above IMO.

File: src/Repository/ProgrammeRepository.php (hunk 1 above)
Comment (live-happy-code-hard) [reply src/Repository/ProgrammeRepository.php:115]: The method call

File: src/Repository/ProgrammeRepository.php (hunk 5 above)
Comment (OvidiuGi) [reply src/Repository/ProgrammeRepository.php:105]: I don't like the way it looks too, but I tried to place the '%' in line 104 but i recieve an error..

**Pull request merged**